src/librustc/hir/mod.rs

```diff
@@ -1616,6 +1616,11 @@ pub enum ExprKind {
     /// and the remaining elements are the rest of the arguments.
     /// Thus, `x.foo::<Bar, Baz>(a, b, c, d)` is represented as
     /// `ExprKind::MethodCall(PathSegment { foo, [Bar, Baz] }, [x, a, b, c, d])`.
+    ///
+    /// To resolve the called method to a `DefId`, call [`type_dependent_def_id`] with
+    /// the `hir_id` of the `MethodCall` node itself.
```

Comment:
This is based on my experiments in https://github.com/rust-lang/rust/pull/66044#discussion_r341932961. I hope this is right.

Reply:
This seems reasonable to me, but I can't speak to if it's the right way to do this.

Reply:
Hm, who would know such things? @Centril @petrochenkov could you have a look?